specs/src/shrex/shrex-sub.md

```diff
@@ -0,0 +1,183 @@
+# ShrEx/Sub Protocol Specification
+
+## Abstract
+
+**ShrEx/Sub** (Share Exchange/Subscribe) is a push-based notification protocol implementing the publish-subscribe pattern for Extended Data Square (EDS) hash distribution in the Celestia data availability network. The protocol enables efficient dissemination of new EDS availability notifications across different node types.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Node Roles](#node-roles)
+- [Message Schema](#message-schema)
+- [Protocol Components](#protocol-components)
+  - [Pubsub Components](#pubsub-component)
+  - [Subscripition Component](#subscription-component)
+- [Message Verification](#message-verification)
+  - [Validation Interface](#validation-interface)
+  - [Validation Pipeline](#validation-pipeline)
+- [Component Interaction Flow](#component-interaction-flow)
+- [Protocol Behavior](#protocol-behavior)
+  - [Publisher Behavior](#publisher-behavior-bn)
+  - [Subscriber Behavior](#subscriber-behavior-ln)
+- [FloodSub vs GossipSub Rationale](#floodsub-vs-gossipsub-rationale)
+  - [Why not GossipSub?](#why-not-gossipsub)
+  - [Why FloodSub?](#why-floodsub)
+- [Additional Notes](#additional-notes)
+- [Links](#links)
+
+## Overview
+
+ShrEx/Sub is built on libp2p's FloodSub router with the following characteristics:
+
+- **Topic ID**: `/eds-sub/0.0.1`
+- **Message Distribution**: Flood-based (sends to all connected peers)
+- **Overlay**: No mesh topology (unlike GossipSub)
+
+### Node Roles
+
+- **Light Nodes (LN)**: Subscribers that receive EDS notifications for data availability sampling
+- **Bridge Nodes (BN)**: Publishers that announce new EDS availability from consensus layer
+
+## Message Schema
+
+The notification message MUST contain the EDS data hash and block height:
+
+```text
+Notification {
+    data_hash: bytes[32]  // EDS root hash
+    height: uint64        // Block height
+}
+```
+
+**Properties:**
+
+- Messages MUST have a fixed 40-byte payload (32 bytes hash + 8 bytes height)
+- Serialization overhead SHOULD be minimal
```

Comment:
Serialisation must be defined in spec

Reply:
Otherwise its like spec saying to itself that it should have a minimal overhead 😁